core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -69,13 +69,24 @@ public void put(PageId pageId, byte[] page) throws IOException {
   }
 
   @Override
-  public ReadableByteChannel get(PageId pageId) throws IOException, PageNotFoundException {
+  public ReadableByteChannel get(PageId pageId, int pageOffset)
+      throws IOException, PageNotFoundException {
+    Preconditions.checkArgument(pageOffset >= 0, "page offset should be non-negative");
     Path p = getFilePath(pageId);
     if (!Files.exists(p)) {
       throw new PageNotFoundException(p.toString());
     }
+    File f = p.toFile();
+    Preconditions.checkArgument(pageOffset <= f.length(),
+        "page offset %s exceeded page size %s", pageOffset, f.length());
     FileInputStream fis = new FileInputStream(p.toFile());
-    return fis.getChannel();
+    try {
+      fis.skip(pageOffset);
+      return fis.getChannel();
+    } catch (Throwable t) {
```

Comment:
nit: is `Throwable` required here to avoid the warning or can we just catch `IOException`?